src/components/plugin-wrapper/hooks/analytics-query-tools.ts

```diff
@@ -0,0 +1,137 @@
+import type { SupportedAxisId } from '@constants/axis-types'
+import type { SupportedInputType } from '@constants/input-types'
+import { getFullDimensionId, isTimeDimensionId } from '@modules/dimension'
+import { getOptionsForRequest } from '@modules/options'
+import { getHeadersMap } from '@modules/visualization'
+import type { CurrentVisualization, DimensionArray, DimensionId } from '@types'
+
+type ParameterRecord = Record<DimensionId, unknown>
+
+export const getRequestOptions = (visualization: CurrentVisualization) => {
+    const options = getOptionsForRequest().reduce(
+        (map, [option, props]): ParameterRecord => {
+            // only add parameter if value !== default
+            if (
+                visualization[option] !== undefined &&
+                visualization[option] !== props.defaultValue
+            ) {
+                map[option] = visualization[option]
+            }
+
+            return map
+        },
+        {} as ParameterRecord
+    )
+
+    return options
+}
```

Comment:
As discussed, this part should probably be improved, but doing so now doesn't make much sense because it ties into the Options feature. So instead of fixing this now we should just ensure this work is included with the Options feature. Not sure if this is a ticket that needs to be created or adjusted.